elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -360,9 +357,18 @@ private String extractOrderBy(Class<?> entityClass, Map<Path, Sorting.SortOrder>
 
                     Path.PathElement last = path.lastElement().get();
 
-                    return FilterPredicate.getTypeAlias(last.getType())
+                    String orderByType = FilterPredicate.getTypeAlias(last.getType())
                             + "."
-                            + getColumnName(entityClass, last.getFieldName())
+                            + getColumnName(entityClass, last.getFieldName());
+
+                    if (last.getType() == entityClass) {
+                        Metric currentMetric = schema.getMetric(getColumnName(entityClass, last.getFieldName()));
```

Comment:
We should not be using the columnName to lookup a metric in the schema.

The columnName is the physical name of the column in the database.  The schema stores things by their logical names.  This will be a bug if the column and schema names don't match.